python/tests/integration/apps/dispatch_probes.py

```
@@ -332,6 +313,59 @@ def readable():
 
 
+@api.get("/t-fd-level-triggered")
+async def t_fd_level_triggered():
+    """A reader that drains only part of the pending bytes must be called
+    again (asyncio's selector is level-triggered; Tokio's reactor is not)."""
+    loop = asyncio.get_running_loop()
+    read_fd, write_fd = os.pipe()
+    done = loop.create_future()
+    chunks = []
+
+    def readable():
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `readable`

Add return type annotation: `None`

(ANN202)